x-pack/plugins/security_solution/server/endpoint/services/actions/create/index.ts

```diff
@@ -113,204 +58,21 @@ export const actionCreateService = (
-        .sign(fleetActionDoc);
-      const signedFleetActionDoc = {
-        ...fleetActionDoc,
-        signed: {
-          data: fleetActionDocSignature.data.toString('base64'),
-          signature: fleetActionDocSignature.signature,
-        },
-      };
-      // write actions to .fleet-actions index
-      try {
-        fleetActionIndexResult = await esClient.index<EndpointAction>(
-          {
-            index: AGENT_ACTIONS_INDEX,
-            body: signedFleetActionDoc,
-            refresh: 'wait_for',
-          },
-          {
-            meta: true,
-          }
-        );
-
-        if (fleetActionIndexResult.statusCode !== 201) {
-          throw new Error(fleetActionIndexResult.body.result);
-        }
-      } catch (e) {
-        // create entry in .logs-endpoint.action.responses-default data stream
-        // when writing to .fleet-actions fails
-        if (doesLogsEndpointActionsDsExist) {
-          await createFailedActionResponseEntry({
-            esClient,
-            doc: {
-              '@timestamp': moment().toISOString(),
-              agent: doc.agent,
-              EndpointActions: {
-                action_id: doc.EndpointActions.action_id,
-                completed_at: moment().toISOString(),
-                started_at: moment().toISOString(),
-                data: doc.EndpointActions.data,
-              },
-            },
-            logger,
-          });
-        }
-
-        throw e;
-      }
-    }
-
-    if (casesClient) {
-      // convert any alert IDs into cases
-      let caseIDs: string[] = payload.case_ids?.slice() || [];
-      if (payload.alert_ids && payload.alert_ids.length > 0) {
-        const newIDs: string[][] = await Promise.all(
-          payload.alert_ids.map(async (alertID: string) => {
-            const cases: CasesByAlertId = await casesClient.cases.getCasesByAlertID({
-              alertID,
-              options: { owner: APP_ID },
-            });
-            return cases.map((caseInfo): string => {
-              return caseInfo.id;
-            });
-          })
-        );
-        caseIDs = caseIDs.concat(...newIDs);
-      }
-      caseIDs = [...new Set(caseIDs)];
-
-      // Update all cases with a comment
-      if (caseIDs.length > 0) {
-        const targets = endpointData.map((endpt: HostMetadata) => ({
-          hostname: endpt.host.hostname,
-          endpointId: endpt.agent.id,
-        }));
-
-        await Promise.all(
-          caseIDs.map((caseId) =>
-            casesClient.attachments.add({
-              caseId,
-              comment: {
-                type: CommentType.actions,
-                comment: payload.comment || '',
-                actions: {
-                  targets,
-                  type: payload.command,
-                },
-                owner: APP_ID,
-              },
-            })
-          )
-        );
-      }
-    }
+    await updateCases({ casesClient, payload, endpointData });
```

Comment:
@paul-tavares I thought a bit more about this and I think the action create service should only be responsible for creating response actions not updating cases using the `caseClient` within. Thus, we should not be updating cases here on line 73, but rather at the API handler hereafter [this](https://github.com/ashokaditya/kibana/blob/25db8cabe42d4cb22d46559c2d7e55e6c73b3cf1/x-pack/plugins/security_solution/server/endpoint/routes/actions/response_actions.ts#L192) try-catch block. 

Maybe we missed this earlier in the code review but if we clean this up then there would be no need to pass the cases client into the action creation service. We should have something like,
```
try {
 await endpointContext.service
        .getActionCreateService()
        .createAction(action);
} catch (e) {
...
...
}

// and update cases after action creation
const endpointData = await endpointContext.service
      .getEndpointMetadataService()
      .getMetadataForEndpoints(esClient, action.endpoint_ids);

await updateCases(action, casesClient, endpointData);
```

If we go this way then we will have to update the file upload handler as well. Maybe we can have a service that updates the cases and then use those in the handlers.

What do you think?

Reply:
Thats sounds fine to me - having the case related logic placed into its own service that we can then call from different locations 👍 

just make sure you get 👀 from @tomsonpl because I believe there will also be impact to Rule automated-response actions (maybe)

Reply:
Thanks @paul-tavares. Yeah, I'll sync with @tomsonpl next week on this PR.

Reply:
Hey, good point! Cases thing should be coupled with action imho. So if you would manager to nicely get it out that would be great :) If it comes to automated actions - we do not depend on cases at all, just `alert_ids` that are both used in cases and automated actions.

Reply:
3db7bf0b577a4395965019ca934e71e9c7983905